devicetypes/sky-nie/min-smart-plug-dimmer.src/min-smart-plug-dimmer.groovy

```diff
@@ -79,23 +89,151 @@ metadata {
     }
 
     preferences {
-        configParams.each {
-            if (it.range) {
-                input "configParam${it.num}", "number", title: "${it.name}:", required: false, defaultValue: "${it.value}", range: it.range
-            } else {
-                input "configParam${it.num}", "enum", title: "${it.name}:", required: false, defaultValue: "${it.value}", options: it.options
+        getConfigParamInput(ledModeParam)
+        getConfigParamInput(autoOffIntervalParam)
+        getConfigParamInput(autoOnIntervalParam)
+        getConfigParamInput(powerFailureRecoveryParam)
+        getConfigParamInput(pushDimmingDurationParam)
+        getConfigParamInput(holdDimmingDurationParam)
+        getConfigParamInput(minimumBrightnessParam)
+        input "disclaimer", "paragraph",
+                title: "WARNING",
+                description: "Configuring for 'Night Light Settings' is only valid for the devices with product number of MP21ZD、MP22ZD、N4002(one of them)",
+                required: false
+        getConfigParamInput(nightLightParam)
+        input "disclaimer", "paragraph",
+                title: "WARNING",
+                description: "Configuring for 'createButton'、'Maximum Brightness' and 'Paddle Control' are only valid for the devices with product number of MS11ZS、MS13ZS、ZW31S、ZW31TS(one of them)",
+                required: false
+        getConfigParamInput(maximumBrightnessParam)
+        getConfigParamInput(paddleControlParam)
+        input(type: "enum", name: "createButton", required: false, title: "Create Button for Paddles?", options: ["No", "Yes"], defaultValue:"Yes")
+    }
+}
+
+private getConfigParamInput(param) {
+    if (param.range) {
+        input "configParam${param.num}", "number", title: "${param.name}:", required: false, defaultValue: "${param.value}", range: param.range
+    } else {
+        input "configParam${param.num}", "enum", title: "${param.name}:", required: false, defaultValue: "${param.value}", options: param.options
+    }
+}
+
+private initialize() {
+    if (device.latestValue("checkInterval") != checkInterval) {
+        sendEvent(name: "checkInterval", value: checkInterval, displayed: false)
+    }
+    if (isButtonAvailable()) {
+        state.createButtonEnabled = (safeToInt(settings?.createButton) != 0)
+        if (state.createButtonEnabled && !childDevices) {
+            try {
+                def child = addChildButton()
+                child?.sendEvent(name: "checkInterval", value: checkInterval, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+            } catch (ex) {
+                log.error("Unable to create button device because the 'Child Button' DTH is not installed",ex)
             }
+        } else if (!state.createButtonEnabled && childDevices) {
+            removeChildButton(childDevices[0])
+        }
+    } else {
+        if (childDevices) {
+            removeChildButton(childDevices[0])
         }
     }
```

Comment:
Wouldn't it look better if You moved that part of the code to a separate function ?
That way You wouldn't have to call whole `initialize()` from `updated()`